nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -128,6 +133,26 @@ public void initialize(String name, Map<String, String> options) {
         catalogOptions);
   }
 
+  private static String inferVersionFromURI(String uri) {
+    if (uri == null) {
+      throw new IllegalArgumentException("URI is not specified in the catalog properties");
+    }
+
+    // match for uri ending with /v1, /v2 etc
+    Pattern pattern = Pattern.compile("/v(\\d+)$");
+    Matcher matcher = pattern.matcher(uri);
+    // Extract and print the version number
+    if (matcher.find()) {
+      return matcher.group(1);
+    } else {
+      throw new IllegalArgumentException(
+          String.format(
+              "URI doesn't end with the version: %s. "
+                  + "Please configure `client-api-version` in the catalog properties explicitly.",
```

Comment:
I wonder whether we can just assume `v2` in this case... but I'm fine with throwing an exception too.

Reply:
If the uri is not ending versions, then it can be a custom implementation which can be based on v1 or v2. So, I didn't want to assume the version there.